packages/flutter/lib/src/widgets/sliver_decoration.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'image.dart';
+
+/// A sliver widget that paints a [Decoration] either before or after its child
+/// paints.
+///
+/// Unlike [DecoratedBox], this widget expects its child to be a sliver, and
+/// must be placed in a widget that expects a sliver. This can allow decorating the
+/// background of a [SliverList] without resorting to a shrink-wrapped list view or
+/// column layout.
```

Comment:
Let's remove mention of the shrinkwrap case. It is an anti-pattern we want to correct in the long term.

Reply:
Done!